model/histogram/float_histogram.go

```diff
@@ -1118,3 +1045,12 @@ func floatBucketsMatch(b1, b2 []float64) bool {
 	}
 	return true
 }
+
+// ReduceResolution reduces the float histogram's spans, buckets into target schema.
+// The target schema must be smaller than the current float histogram's schema.
+func (h *FloatHistogram) ReduceResolution(targetSchema int32) *FloatHistogram {
+	h.PositiveSpans, h.PositiveBuckets = reduceResolution(h.PositiveSpans, h.PositiveBuckets, h.Schema, targetSchema, false)
+	h.NegativeSpans, h.NegativeBuckets = reduceResolution(h.NegativeSpans, h.NegativeBuckets, h.Schema, targetSchema, false)
+
+	return h
```

Comment:
@fatsheep9146 / @beorn7 shouldn't there be `h.Schema = targetSchema`? While the comment only speaks about spans and buckets, name and the API of this method would make me expect that the whole histogram gets reduced as such.
Same question applies to the integer histogram below.

Reply:
Thanks! I think you are right, I will fix this and add more test about this function.

Reply:
@beorn7 @linasm #13116 is created